src/main/java/com/google/sps/data/Room.java

```diff
@@ -0,0 +1,150 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.Entity;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.List;
+import java.util.HashMap;
+import javax.servlet.http.HttpServletRequest;
+import com.google.appengine.api.datastore.Key;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import java.util.stream.Collectors;
+import java.util.Map;
+import java.util.stream.*;
+
+
+public class Room {
+    private static final String ROOM_ENTITY = "Room";
+    private static final String MEMBERS_PROPERTY = "members";
+    private static final String MESSAGES_PROPERTY = "messages";
+    private static final String VIDEOS_PROPERTY = "videos";
+    private static DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    private List<Member> members;
+    private LinkedList<Message> messages;
+    private Queue<Video> videos;
+
+    /**
+      * Room constructor
+      * @param members a List of Member objects
+      * @param videos an Queue of video objects
+      * @return a new Room object
+      */
+    private Room(List<Member> members, Queue<Video> videos){
+        this.messages = new LinkedList<Message>();
+        this.members = members;
+        this.videos = videos;
+    }
+
+    /**
+      * Room constructor
+      * @param members a List of Member objects
+      * @param videos an Queue of video objects
+      * @param messages a List of message objects
+      * @return a new Room object
+      */
+    private Room(List<Member> members, Queue<Video> videos, LinkedList<Message> messages) {
+        this.members = members;
+        this.videos = videos;
+        this.messages = messages;
+    }
+
+    //Room factory function
+    public static Room createRoom(List<Member> members, Queue<Video> videos, LinkedList<Message> messages){
+        return new Room(members, videos, messages);
+    }
+
+    //Returns the Room's members
+    public List<Member> getMembers() {
+        return this.members;
+    }
+
+    //Get all of the members as a list of EmbeddedEntities
+    private List<EmbeddedEntity> getMembersAsEmbeddedEntities() {
+        return (List<EmbeddedEntity>) members.stream().map(Member::toEmbeddedEntity).collect(Collectors.toList());
+    }
+
+    //Returns the Room's video url list
+    public Queue<Video> getVideos() {
+        return this.videos;
+    }
+
+    //Returns the Room's message list
+    public LinkedList<Message> getMessages(){
+        return this.messages;
+    }
+
+    //Returns the messages as a list of embedded entities
+    private List<EmbeddedEntity> getMessagesAsEntities() {
```

Comment:
order your function by type and access level. static function first and then member function, inside, public first and then private.